src/main.cpp

```diff
@@ -66,8 +66,10 @@ BlockMap mapBlockIndex;
 CChain chainActive;
 CBlockIndex *pindexBestHeader = NULL;
 static std::atomic<int64_t> nTimeBestReceived(0); // Used only to inform the wallet of when we last received a block
-CWaitableCriticalSection csBestBlock;
-CConditionVariable cvBlockChange;
+CWaitableCriticalSection g_best_block_mutex;
+CConditionVariable g_best_block_cv;
+uint256 g_best_block;
```

Comment:
Since this adds a couple new global variables, please document them thoroughly, including where they are intended to be used. It'd be even nicer if we could figure out a way to avoid them being globals, but I don't see any obvious value that they could be tied to instead, so documentation will have to suffice.

Reply:
I think cherry-picked commits should be as close to the source commits as possible (minimal diff) so that reviewers can see that they're at least functionally identical; then we can make further improvements (for example, in this case, not having these be global) in a separate PR. However, I do agree that adding documentation to a cherry-pick (beyond what's in the original commit) is fine; I'll do that here, thanks for the suggestion.

Reply:
I agree; can you add those as a separate commit?